src/pages/VenueLandingPage/VenueLandingPage.tsx

```diff
@@ -81,19 +87,44 @@ export const VenueLandingPage: React.FunctionComponent<VenueLandingPageProps> =
   );
 
   dayjs.extend(advancedFormat);
-  const [isPaymentModalOpen, setIsPaymentModalOpen] = useState(false);
+  const [isPaymentModalOpen, setPaymentModalOpen] = useState(false);
   const [selectedEvent, setSelectedEvent] = useState<
     WithId<VenueEvent> | undefined
   >();
-  const [isAuthenticationModalOpen, setIsAuthenticationModalOpen] = useState(
+  const [isAuthenticationModalOpen, setAuthenticationModalOpen] = useState(
     false
   );
+  const [isJoinVenueModalOpen, setJoinVenueModalOpen] = useState(false);
   const [shouldOpenPaymentModal, setShouldOpenPaymentModal] = useState(false);
   const [eventPaidSuccessfully, setEventPaidSuccessfully] = useState<
     string | undefined
   >();
 
-  const { user } = useUser();
+  const closePaymentModal = () => {
+    setPaymentModalOpen(false);
+  };
+
+  const openAuthenticationModal = () => {
+    setAuthenticationModalOpen(true);
+  };
+
+  const closeAuthenticationModal = () => {
+    setAuthenticationModalOpen(false);
+  };
+
+  const showJoinVenueModal = useCallback(() => {
+    setJoinVenueModalOpen(true);
+  }, []);
+
+  const hideJoinVenueModal = useCallback(() => {
+    setJoinVenueModalOpen(false);
+  }, []);
+
+  const handlePasswordSubmit = () => {
+    if (user) return;
+
+    openAuthenticationModal();
+  };
```

Comment:
`useCallback`'s for each of these

Reply:
[OOS] We seem to use this `useState` with a boolean, then `useCallback`'d 'setTrue`, `setFalse` pattern a fair bit.

We could make a `useBooleanState` or `useToggleState`  or similar that wraps up this pattern for us. I would probably make it return an array of 3 things. The state, the setTrue, and the setFalse function (using an array so that people can choose whatever name they want for the functions in a similar way to how we currently use `useState`)